File: src/main/java/no/nav/fo/veilarbregistrering/kafka/formidlingsgruppe/InsertFormidlingsgruppeMapper.java
package no.nav.fo.veilarbregistrering.kafka.formidlingsgruppe;

import no.nav.fo.veilarbregistrering.arbeidssoker.Formidlingsgruppe;
import no.nav.fo.veilarbregistrering.kafka.FormidlingsgruppeEvent;

class InsertFormidlingsgruppeMapper extends FormidlingsgruppeMapper {

    @Override
    protected FormidlingsgruppeEvent map(GgArenaFormidlinggruppeDto ggArenaFormidlinggruppeDto) {
        AfterDto after = ggArenaFormidlinggruppeDto.getAfter();

        return new FormidlingsgruppeEvent(
                mapFoedselsnummer(after.getFODSELSNR()),
                after.getPERSON_ID(),
                after.getPERSON_ID_STATUS(),
                mapOperation(ggArenaFormidlinggruppeDto.getOp_type()),
                Formidlingsgruppe.of(after.getFORMIDLINGSGRUPPEKODE()),
                modDato(after.getMOD_DATO()),
                null,
                null);
    }
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/kafka/formidlingsgruppe/InsertFormidlingsgruppeMapper.kt
package no.nav.fo.veilarbregistrering.kafka.formidlingsgruppe

import no.nav.fo.veilarbregistrering.arbeidssoker.Formidlingsgruppe
import no.nav.fo.veilarbregistrering.arbeidssoker.formidlingsgruppe.FormidlingsgruppeEvent

internal class InsertFormidlingsgruppeMapper : FormidlingsgruppeMapper() {
    override fun map(ggArenaFormidlinggruppeDto: GgArenaFormidlinggruppeDto): FormidlingsgruppeEvent {
        val after = ggArenaFormidlinggruppeDto.after ?: throw InvalidFormidlingsgruppeEvent("Insert operation requires non null after-section")
        return FormidlingsgruppeEvent(
            mapFoedselsnummer(after.FODSELSNR),
            after.PERSON_ID,
            after.PERSON_ID_STATUS,
            mapOperation(ggArenaFormidlinggruppeDto.op_type),
            Formidlingsgruppe(after.FORMIDLINGSGRUPPEKODE),
            modDato(after.MOD_DATO),
            null,
            null
        )
    }
}
